sdks/java/io/cdap/build.gradle

```diff
@@ -52,14 +52,17 @@ dependencies {
     implementation library.java.cdap_plugin_zendesk
     implementation library.java.commons_lang3
     implementation library.java.guava
+    implementation library.java.google_code_gson
     implementation library.java.hadoop_common
     implementation library.java.hadoop_mapreduce_client_core
     implementation library.java.jackson_core
     implementation library.java.jackson_databind
     implementation library.java.slf4j_api
+    implementation library.java.spark_streaming
```

Comment:
Just noticing this is based on Spark 2.4 and wondering.... considering Spark 2 is close to reaching it's end of life, wouldn't it make more sense to kick this of with Spark 3?

Reply:
Good question, I missed that

Reply:
Yes, it would be good to clarify that.
We don't use Spark environment in CdapIO. Our goal is only to mimic all Spark classes related to Cdap Plugins in order to use the business logic.
At the moment, all the Cdap plugins we want to support use Spark 2, so we use Spark 2 here.

Reply:
@mosche What do you mean by naming scheme?
Do you suggest to add something like this?
```
def shouldProvideSpark2 = true
def shouldProvideSpark3 = false
```
...
```
if (shouldProvideSpark2) {
    implementation library.java.spark_streaming
    implementation library.java.spark_core
}
if (shouldProvideSpark3) {
    implementation library.java.spark3_streaming
    implementation library.java.spark3_core
}
```

Reply:
@Amar3tto Unfortunately this can't be dynamically resolved. If we (eventually) want to support Spark 2 and 3, there has to be 2 maven artifacts. I was talking about a naming scheme for these.
So something like:
- org.apache.beam/beam-sdks-java-io-cdap-spark2
- org.apache.beam/beam-sdks-java-io-cdap-spark3

Reply:
I want to clarify: within the current project, support for Spark version 3 is a "[non-goal](https://docs.google.com/document/d/1T-bhd0Qk7DBePIfgHEPagYiA1oLP4z5kYEd0S1SOGxQ/edit#heading=h.cf98cc3ttmdn)".

Reply:
> I want to clarify: within the current project, support for Spark version 3 is a "[non-goal](https://docs.google.com/document/d/1T-bhd0Qk7DBePIfgHEPagYiA1oLP4z5kYEd0S1SOGxQ/edit#heading=h.cf98cc3ttmdn)".

👍 my primary concern was the appropriate naming of artifacts to make the Spark dependency obvious and allow adding support for Spark 3 later

Reply:
Shouldn't we apply the same naming strategy as we do for Spark dependencies of versions 2 and 3 for Spark runner? I know it's not perfect but, at least, it won't be more confusing.

Reply:
@mosche do you mean ":sdks:java:io:sparkreceiver" should be renamed to may be ":sdks:java:io:spark2receiver" ?

If so, +1 for that.

Reply:
yes, modules that depends on spark should contain the spark version in their name. spark2receiver sounds good 👍 
It looks like the cdap module also depends on spark, so the same applies there:
```
    implementation library.java.spark_streaming
```

Reply:
Unfortunately, I'm disagree with that, at least, for Spark v2 entities. We don't have this naming strategy for Spark runner (why? it's another question), so it will be confusing to have `spark2*` and `spark*` names in the same time. 

Though, for Spark v3 we used to have `spark3` prefix, so we can continue that strategy for `spark4` and etc if it will be needed. 

Actually, it looks as more wide question that can be applied for similar cases in other Beam components.

Reply:
Hmm, I personally don't see why it's necessary to repeat the same mistake just because it was made once? 
Also IOs and runners are very different in nature. Replacing the runner is trivial, but with the Cdap IO you're better aware right from the start that it depends on Spark 2 (and Spark 3 is not (yet) supported). 
In any case, I don't wanna block this PR ... just a long hanging fruit and missed opportunity in my eyes.

Reply:
Well, then it should be `:sdks:java:io:sparkreceiver:2` (like `:runners:spark:2` and `:runners:spark:3`) but not `:sdks:java:io:spark2receiver`

Reply:
Yeah, ":sdks:java:io:sparkreceiver:2" seems to be more consistent with the current convention.

Reply:
Thank you all for the review and comments on the PR. Can we do this renaming you're talking about in the next PR (after this one is completed) without running into multiple conflicts?

Reply:
👍

Reply:
@Lizzfox If would be great to make these changes (I'm fine for a separate PR for this) till the next release branch cut.

Reply:
@aromanenko-dev got it, thanks! We'll make changes as soon as this PR is merged and I'll tag you in the comments.